usr/sbin/pkcstok_migrate/pkcstok_migrate.c

```diff
@@ -2491,27 +2491,19 @@ static CK_BBOOL pkcsslotd_running(void)
         /* if endptr is not a null character, the directory is not
          * entirely numeric, so ignore it */
         long lpid = strtol(ent->d_name, &endptr, 10);
-        if (*endptr != '\0') {
+        if (*endptr != '\0')
             continue;
-        }
 
-        /* try to open the cmdline file */
+        /* try to open the cmdline file. Ignore it if open fails */
         snprintf(fname, sizeof(fname), "/proc/%ld/cmdline", lpid);
         fp = fopen(fname, "r");
-        if (!fp) {
-            warnx("fopen(%s) failed, errno=%s", fname, strerror(errno));
-            return CK_TRUE;
-        }
+        if (!fp)
```

Comment:
I guess this is mostly the case with SELINUX or other LSMs active.  But in general, couldn't we simpllify this whole function a lot?  I mean, we do have a PID file for a running pkcsslotd.  So we could just use that file:
1. If it does not exist => pkcsslotd is not running (yes, a user might have deleted it while pkcsslotd is still running, but then we do not care...)
2. If the PID mentioned in the file does not exist => pkcsslotd is not running (PID file is stale)
3. If /proc/<PID>/cmdline cannot be opened => Assume pkcsslotd is running
4. If /proc/<PID>/cmdline contains the pkcsslotd command => pkcsslotd is running
5. else pkcsslotd is not running.

Sounds easier than iterating through the whole /proc directory skipping all the files in there that are not directories for running processes.

Reply:
Changed the code to use the pid file. Please take a took again.